test/GogeDao.t.sol

```diff
@@ -319,383 +455,792 @@ contract DaoTest is Utility, Test {
+        assertEq(activePolls.length, 3);
+        assertEq(activePolls[0], 1);
+        assertEq(activePolls[1], 2);
+        assertEq(activePolls[2], 3);
 
-        // Verify state change
-        assertEq(gogeDao.pollNum(), 1);
-        assert(gogeDao.pollTypes(1) == GogeDAO.PollType.funding);
+        assertEq(gogeToken.balanceOf(address(joe)), 10_000_000_000 ether);
+        assertEq(gogeToken.balanceOf(address(gogeDao)), gogeDao.minAuthorBal() * 3);
 
-        // Verify poll metadata
-        assertEq(gogeDao.getMetadata(1).description, "I want to propose a funding");
-        assertEq(gogeDao.getMetadata(1).time1, block.timestamp);
-        assertEq(gogeDao.getMetadata(1).time2, block.timestamp + 2 days);
-        assertEq(gogeDao.getMetadata(1).addr1, address(joe));
-        assertEq(gogeDao.getMetadata(1).addr2, BUSD);
-        assertEq(gogeDao.getMetadata(1).amount, 1_000 ether);
+        // Joe places votes on poll 1
+        assert(joe.try_approveToken(address(gogeToken), address(gogeDao), 10_000_000_000 ether));
+        assert(joe.try_addVote(address(gogeDao), 1, 10_000_000_000 ether));
 
-        // NOTE pass poll
-        
-        uint256 joe_votes = 50_000_000_000 ether;
-        gogeDao.updateVetoEnabled(false);
+        //Warp to end time
+        vm.warp(block.timestamp + 2 days);
 
-        mint("BUSD", address(gogeDao), 1_000 ether);
+        // Call queryEndTime()
+        gogeDao.queryEndTime();
+
+        // Post-state check.
+        assertEq(gogeDao.passed(1),                     false);
+        assertEq(gogeToken.isBlacklisted(address(joe)), false);
+        assertEq(gogeDao.getMetadata(1).endTime, block.timestamp);
+
+        activePolls = gogeDao.getActivePolls();
+        assertEq(activePolls.length, 2);
+        assertEq(activePolls[0], 3);
+        assertEq(activePolls[1], 2);
+
+        assertEq(gogeToken.balanceOf(address(joe)), 10_000_000_000 ether);
+        assertEq(gogeToken.balanceOf(address(gogeDao)), gogeDao.minAuthorBal() * 2);
+    }
+
+    /// @notice Verify execution of poll when an admin calls passPoll
+    function test_gogeDao_passPoll() public {
+        test_gogeDao_createPoll();
+        gogeDao.setGateKeeping(false);
+        gogeDao.transferOwnership(address(dev));
 
         // Pre-state check.
-        assertEq(IERC20(BUSD).balanceOf(address(joe)), 0);
-        assertEq(IERC20(BUSD).balanceOf(address(gogeDao)), 1_000 ether);
         assertEq(gogeDao.passed(1), false);
-        assertEq(gogeDao.pollEndTime(1), block.timestamp + 2 days);
+        assertEq(gogeDao.isActivePoll(1), true);
+        assertEq(gogeToken.isBlacklisted(address(joe)), false);
+        assertEq(gogeDao.getMetadata(1).endTime, block.timestamp + 2 days);
+
+        // passPoll
+        assert(dev.try_passPoll(address(gogeDao), 1));
+
+        // Post-state check.
+        assertEq(gogeDao.passed(1), true);
+        assertEq(gogeDao.isActivePoll(1), false);
+        assertEq(gogeToken.isBlacklisted(address(joe)), true);
```

Comment:
I removed this. I was checking this before because when I was calling test_gogeDao_createPoll instead of create_mock_poll; That poll it was creating was to blacklist Joe. So checking for blacklist was the same as me checking to see if the poll has been executed, which was pointless because I can just call gogeDao.passed(pollNum).